Sources/Task/ResultPromise.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  ResultPromise.swift
+//  Deferred
+//
+//  Created by Zachary Waldowski on 10/27/15.
+//  Copyright © 2015-2016 Big Nerd Ranch. Licensed under MIT.
+//
+
+#if SWIFT_PACKAGE
+import Deferred
+import Result
+#endif
+
+extension PromiseType where Value: ResultType {
+    /// Completes the task with a successful `value`, or a thrown error.
+    ///
+    /// - seealso: `fill(_:)`
+    public func succeed(@autoclosure value: () throws -> Value.Value) -> Bool {
+        return fill(Value(value: value))
+    }
```

Comment:
Like the `ResultType` initializer, I don't think this works like we want it to. Need to test if `future.succeed(try foo())` (or `try future.succeed(foo())`?) still requires the surrounding context be `throws`.